### PR TITLE
PutSecureParameter and recursive GetAllParametersByPath

### PR DESCRIPTION
**Pull request by ingshtrom**
* Add easier way to set secure SSM parameters
* recursively get parameters for a given path. By recursive, I mean that it gets more than the limit of 10 parameters per invocation, which is an AWS API limit.

### CODE REVIEW

**Comment by geototti21** [comment]
👋 hey @ingshtrom, thanks for your contribution 😄 . The limit on params will be fixed on #29. Can you please refactor your PR to include only the put parameter feature? 🙌

**Comment by ingshtrom** [comment]
@geototti21 of course! I subscribed to #29 so I'll wait until it's merged to fully integrate. Thank you!

**Comment by geototti21** [comment]
> @geototti21 of course! I subscribed to #29 so I'll wait until it's merged to fully integrate. Thank you!

Thank you! check our new release [v0.6.0](https://github.com/PaddleHQ/go-aws-ssm/releases/tag/v0.6.0) please

**Comment by ingshtrom** [comment]
ready for review, PTAL @geototti21.

Thank you!

File: parameter_store_client.go
Comment (geototti21) [inline parameter_store_client.go:129]: Based on their docs [here](https://docs.aws.amazon.com/sdk-for-go/api/service/ssm/#SSM.PutParameter) I think the error code you want to check is `ssm. ParameterAlreadyExists` and also return `ErrParameterInvalidName` error

File: parameter_store_client_test.go
Comment (geototti21) [inline parameter_store_client_test.go:118]: This case is already covered in the first "Success" test case, maybe amend the other test?

File: parameter_store_client_test.go
Comment (geototti21) [inline parameter_store_client_test.go:22]: please move this global variable inside the `stubSSMClient` struct like we do with the other methods and try to access it from there to assert

**Review by geototti21 — CHANGES_REQUESTED** [review]
left you some comments but looks good

File: parameter_store_client_test.go
Comment (ingshtrom) [reply parameter_store_client_test.go:22]: I haven't done a _whole lot_ of Go, so bear with me here, please. When I put the variable inside of the `stubSSMClient`, I can write to it fine within `stubSSMClient.PutParameter()` (L65), but then when I try to access it anywhere afterwards, it is not available because the interface is more restrictive (`ssmClient` only has 3 methods defined for the interface) or I can cast the type, but then we lose the extra parts defined by the `stubSSMClient`, e.g. `stubSSMClient.PutParameterInputReceived`. 

I will updated the branch with what I have so far, though.

**Comment by ingshtrom** [comment]
now I'm just using the stub client directly in the test object, but it still isn't working and now I'm just plain confused 😕

**Comment by geototti21** [comment]
@ingshtrom apologies for the late reply, I pushed this [commit](https://github.com/PaddleHQ/go-aws-ssm/pull/27/commits/2f6004880a5cb1ae7e6b8bdf4f8e609632d48fbe) that fixed your tests. It was more of a bigger problem in the tests that we didn't use a pointer receiver in the stub client, apologies for that.

**Review by geototti21 — APPROVED** [review]
Thanks for this @ingshtrom, I also tested this locally, and works perfectly 🚀

**Pull request merged**
